### PR TITLE
PM-11123: Device Type mapping

### PR DESCRIPTION
**Pull request by mzieniukbw**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11123

## 📔 Objective

- Device Type to Client Type mappings. (Split from #4741)
- Moved `ClientType` from `Bit.Core.NotificationCenter.Enums` namespace to `Bit.Core.Enums` namespace.
- Renamed `ClientType` from `Bit.Core.Identity` namespace to `IdentityClientType` to remove the naming clash and make it more clear on it's purpose.
- Rename `ClientType` in `ICurrentContext` to `IdentityClientType`

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by withinfocus — COMMENTED** [review]
I see the `DeviceTypeToClientType` method in your other PR being useful here. The client type can be lifted out of the NotificationCenter namespace as well, and then the other classes we spoke about being cleaned up too.

**Comment by mzieniukbw** [comment]
@withinfocus i wanted to create a follow up PR once this is done with the unification of Client Types and mapping, but i can fit it into one PR too.

**Comment by withinfocus** [comment]
Whatever is easiest. That one method is something I will use for https://bitwarden.atlassian.net/browse/PM-11984.

File: src/Core/Utilities/DeviceTypes.cs [hunk 1]
```diff
@@ -4,21 +4,56 @@ namespace Bit.Core.Utilities;
...
+
+    private static ClientType ToClientType(DeviceType? deviceType)
+    {
+        return deviceType switch
+        {
+            not null when MobileTypes.Contains(deviceType.Value) => ClientType.Mobile,
+            not null when DesktopTypes.Contains(deviceType.Value) => ClientType.Desktop,
+            not null when BrowserExtensionTypes.Contains(deviceType.Value) => ClientType.Browser,
+            not null when BrowserTypes.Contains(deviceType.Value) => ClientType.Web,
+            _ => ClientType.All
```
Comment (withinfocus) [inline src/Core/Utilities/DeviceTypes.cs:56]: 💭 Not sure about this being the best to return as a default, but I don't have a better answer besides `null` and we don't want to deal with nullability here.

**Review by withinfocus — APPROVED** [review]
Looking good! I asked around and we couldn't think of a better name than `IdentityClientType` so that works, at least for now.

**Review by differsthecat — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Pull request merged**